### PR TITLE
Last.fm integration

### PR DESCRIPTION
**Pull request by dotted**
My initial attempt at Last.fm integration, just looking for a review of the code so far to see if I'm on the right track.

`!song` and `!earliersong` (aliases of this has also been added ie. `!lastsong`) has been implemented, but only lightly tested.

### CODE REVIEW

File: lib/commands/implementations/song.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const Command = require('../command-interface');
+const CommandOutput = require('../command-output');
+
+function getCurrentPlayingSongForBroadcaster(input, services) {
+  return services.lastfm.getCurrentPlayingSong()
+    .then((song) => {
+      const profilePageUrl = services.lastfm.getProfilePage();
+      if (!song) {
```
Comment (linusred) [inline lib/commands/implementations/song.js]: Looks like we return a null here, so let's compare to a null.

File: lib/services/lastfm.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        return {
+          trackName,
+          artistName,
+          playedAgo,
+          nowPlaying,
+        };
+      }
+      return null; // `No song has been scrobbled to ${this.getProfilePage()}`;
+    }).catch((err) => {
+      console.log(err);
```
Comment (linusred) [inline lib/services/lastfm.js]: Only thing left to do from my perspective here is maybe a unit test with a mocked fetcher to test that objects are parsin' correctly.

And then get rid of these catches so they can be handled by the downstream callers.

**Comment by dotted** [comment]
Ping @linusred

**Comment by dotted** [comment]
Ping @linusred

**Comment by linusred** [comment]
:shipit:

**Pull request merged**